### PR TITLE
Roll Call

### LINKED ISSUE #172: Add roll-call feature
Pets are currently assigned names from a list, but there is now way to see what pets you have and what their names are (other than the remove list and the friendship announcements)

We need an action for a roll-call where each pet announces themselves using the notifications channel. 

This also opens up the testing for custom names

### PR DESCRIPTION
**Pull request by Harry-Hopkinson**
Fixes #172

### CODE REVIEW

**Comment by Harry-Hopkinson** [comment]
@tonybaloney What format would you like to be printed out? Currently I have `Type ${ pet.type }, Name ${pet.pet.name()}, Color ${ pet.color } Assigned for Duty!`.

**Comment by Harry-Hopkinson** [comment]
I have updated the main-bundle by running npm run compile. It is just code from other people that has not been built.

File: package.json [hunk 1]
```diff
@@ -101,6 +101,11 @@
                     "dark": "media/icon/dark-trash.svg",
                     "light": "media/icon/light-trash.svg"
                 }
+            },
+            {
+                "command": "vscode-pets.rollcall",
+                "title": "Rollcall",
```
Comment (tonybaloney) [inline package.json:107]: Hyphenate to "Roll-call"

File: package.json [hunk 2]
```diff
@@ -101,6 +101,11 @@
                     "dark": "media/icon/dark-trash.svg",
                     "light": "media/icon/light-trash.svg"
                 }
+            },
+            {
+                "command": "vscode-pets.rollcall",
```
Comment (tonybaloney) [inline package.json:106]: Change the command to `roll-call` as well so it's consistent with the backend

File: media/main-bundle.js [hunk 3]
```diff
@@ -486,6 +600,16 @@ function petPanelApp(basePetUri, theme, themeKind, petColor, petSize, petType) {
...
                 });
                 break;
+            case 'roll-call':
+                var pets = exports.allPets.pets();
+                // go through every single
+                // pet and then print out their name
+                pets.forEach((pet) => {
+                    vscode.postMessage({
+                        command: 'info',
+                        text: `Type ${pet.type}, Name ${pet.pet.name()}, Color ${pet.color} Assigned for Duty!`,
```
Comment (tonybaloney) [inline media/main-bundle.js:610]: `${pet.pet.emoji()} ${pet.pet.name()} (${pet.color}: ${pet.pet.hello()}`

Add a `.hello()` method to the pet interface so each pet can have a custom hello message.

File: src/extension/extension.ts [hunk 4]
```diff
@@ -360,33 +360,18 @@ export function activate(context: vscode.ExtensionContext) {
...
-                    collection.push(spec);
-                    storeCollectionAsMemento(context, collection);
-                }
+                createPetPlayground(context);
+            }
+        }),
+    );
+
+    context.subscriptions.push(
+        vscode.commands.registerCommand('vscode-pets.rollcall', async () => {
```
Comment (tonybaloney) [inline src/extension/extension.ts:369]: Update to `roll-call`

File: src/panel/main.ts [hunk 5]
```diff
@@ -390,6 +390,21 @@ export function petPanelApp(
                         .join('\n'),
                 });
                 break;
+
+            case 'roll-call':
+                var pets = allPets.pets();
+                // go through every single
+                // pet and then print out their name
+                pets.forEach((pet) => {
```
Comment (tonybaloney) [inline src/panel/main.ts:398]: As above (see custom message)

**Review by tonybaloney — CHANGES_REQUESTED** [review]

**Review by tonybaloney — APPROVED** [review]

**Comment by tonybaloney** [comment]
Great contribution, thank you

**Pull request merged**